src/main/kotlin/reactor/kotlin/extra/math/MathFluxExtensions.kt

```diff
@@ -33,6 +34,45 @@ import java.util.function.Function
  */
 fun <T: Number> Flux<T>.sum(): Mono<Long> = MathFlux.sumLong(this)
 
+/**
+ * Extension to compute the [Int] sum of all values emitted by a [Flux] of [Number]
+ * and return it as a [Mono] of [Int].
+ *
+ * Note that numbers will be mapped to [Int] using Java standard conversions
+ * This may lead to arbitrary precision gain or loss if the source flux contains different types
+ * Please choose an appropriate method based on the expected types in the source flux.
+ *
+ * @author Mark Pruden
+ * @since 1.1.5
+ */
+fun <T: Number> Flux<T>.sumInt(): Mono<Int> = MathFlux.sumInt(this)
```

Comment:
would there be compiler ambiguity if we kept `<T:Number> sum` above AND turn these additional methods into `<T:Int>` `<T:Long>` ?
my goal was to use Kotlin as much as possible to avoid the "dumb" style of the Java `MathFlux` (which repeats the type in the method name).

Reply:
Totally get where you are coming from. Your suggestion Does create a compiler Conflict with the existing `sum(): Mono<Long>` method

However there is potentially a workable solution which achieves the outcome. But it requires replacing the existing `sum(): Mono<Long>` method with the below method.

```
inline fun <reified T: Number> Flux<T>.sum(): Mono<T> =
    when (T::class) {
        BigDecimal::class -> MathFlux.sumBigDecimal(this) as Mono<T>
        Double::class -> MathFlux.sumDouble(this) as Mono<T>
        Long::class -> MathFlux.sumLong(this) as Mono<T>
        // todo above cases need to be exhaustive
        else -> Mono.empty() // or error()
    }
```

Disadvantages 
1) Existing users of sum() may be broken since new function returns Mono<T> (not Mono<Long>) - so existing code would would need to explicitly migrate to sumLong() - OR NOT - depending on use of resultant Mono<T>

Advantages
1) The sum() method would work for all cases (probably majority) where the input types holds enough precision, and there is no expectation of type conversion.
2) The new sum() would always return the same type that was passed in so there is no ambiguity. eg. `fluxOfInts.sum()` - makes no statement about output type so user can assume same type.
3) If the user needs to increase precision or change output type then the full method e.g. e.g. `fluxOfInts.sumLong()` - makes an Explicit statement about the (output) type of result that is needed.

Reply:
I very much like that 🤩

Perhaps the same style could be applied to the map-and-sum/average case ?
And there, since the mapping is explicit, no need for individual per-type methods at all!

Regarding the disadvantage, even if I'd argue that it is more desirable for the `sum` method to maintain the type, it is indeed a breaking compatibility change 😞

can we get creative with the name and deprecate the old `sum()` perhaps?

in Reactor we have the `collect()` and `reduce()` operators.
So perhaps something like `collectSum()`/`collectAverage()`?

For the map case, `mapAndCollectSum()` and `mapAndCollectAverage()`?

Any good candidates for names from the Kotlin world?

Reply:
> Perhaps the same style could be applied to the map-and-sum/average case ?
Absolutely that was the intent.

> can we get creative with the name and deprecate the old sum() perhaps?
or make the breaking change in a major version release.

> can we get creative with the name and deprecate the old sum() perhaps?
Deprecation would be a good idea in any case.

> So perhaps something like collectSum()/collectAverage()?
Fo these methods I dont have any better idea's than what you suggest

> Any good candidates for names from the Kotlin world?
In kotlin world found the sumOf() method in sequence's where a mapping function exists, suggest this would be reasonable alternative, and is shorter.

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.sequences/sum-of.html

thus
```
Flux<String> fluxOfSentences;
Mono<Double> numberOfCharsDouble = fluxOfSentences.sumOf(s -> (double) s.length());
Mono<Int> numberOfChars = fluxOfSentences.sumOf(String::length)
Mono<Int> averageWordsPerSentence = fluxOfSentences.averageOf(s -> s.split(" ").length);
```

As for the non-mapping functions, suggest go with your suggestion until future major release where the deprecared sum() average() functions potentially could be replaced.

Reply:
Ah, sumAll might be a good one too